File: wrappers/java/src/test/java/org/hyperledger/indy/sdk/non_secrets/GetRecordTest.java
package org.hyperledger.indy.sdk.non_secrets;

import org.hyperledger.indy.sdk.wallet.WalletItemNotFoundException;
import org.json.JSONObject;
import org.junit.Test;

import java.util.concurrent.ExecutionException;

import static org.hamcrest.CoreMatchers.isA;
import static org.junit.Assert.assertEquals;
import static org.junit.Assert.assertTrue;

import org.hyperledger.indy.sdk.JsonObjectSimilar;

public class GetRecordTest extends NonSecretsIntegrationTest {

	@Test
	public void testGetRecordWorksForDefaultOptions() throws Exception {
		WalletRecord.add(wallet, type, id, value, tags).get();

		String recordJson = WalletRecord.get(wallet, type, id, optionsEmpty).get();

		JSONObject actual = new JSONObject(recordJson);

		JSONObject expected = new JSONObject()
				.put("id", id)
				.putOpt("type", JSONObject.NULL)
				.put("value", value)
				.put("tags", JSONObject.NULL);

		assertTrue(JsonObjectSimilar.similar(expected, actual));

	}

	@Test
	public void testGetRecordWorksForFullData() throws Exception {
		WalletRecord.add(wallet, type, id, value, tags).get();

		String optionsJson = "{" +
				"   \"retrieveType\": true,\n" +
				"   \"retrieveValue\": true,\n" +
				"   \"retrieveTags\": true" +
				"}";
		String recordJson = WalletRecord.get(wallet, type, id, optionsJson).get();

		JSONObject record = new JSONObject(recordJson);
		assertEquals(id, record.getString("id"));
		assertEquals(type, record.getString("type"));
		assertEquals(value, record.getString("value"));
		assertTrue(JsonObjectSimilar.similar(new JSONObject(tags), record.getJSONObject("tags")));
	}

	@Test
	public void testGetRecordWorksForNotFoundRecord() throws Exception {
		thrown.expect(ExecutionException.class);
		thrown.expectCause(isA(WalletItemNotFoundException.class));

		WalletRecord.get(wallet, type, id, optionsEmpty).get();
	}
}

File: wrappers/dotnet/indy-sdk-dotnet-test/NonSecretsTests/GetRecordTest.cs
using Hyperledger.Indy.NonSecretsApi;
using Hyperledger.Indy.WalletApi;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Newtonsoft.Json;
using Newtonsoft.Json.Linq;
using System.Threading.Tasks;

namespace Hyperledger.Indy.Test.NonSecretsTests
{
    [TestClass]
    public class GetRecordTest : NonSecretsIntegrationTestBase
    {
        [TestMethod]
        public async Task TestGetRecordWorksForDefaultOptions()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);

            var recordJson = await NonSecrets.GetRecordAsync(wallet, type, id, optionsEmpty);            
            var actual = JObject.Parse(recordJson);

            var expected = JObject.FromObject(new
            {
                id = id,
                type = (object)null,
                value = value,
                tags = (object)null
            });

            Assert.IsTrue(JValue.DeepEquals(expected, actual));
        }

        [TestMethod]
        public async Task TestGetRecordWorksForFullData()
        {
            await NonSecrets.AddRecordAsync(wallet, type, id, value, tags);

            var optionsJson = JsonConvert.SerializeObject(
                new
                {
                    retrieveType = true,
                    retrieveValue = true,
                    retrieveTags = true
                }
            );

            var recordJson = await NonSecrets.GetRecordAsync(wallet, type, id, optionsJson);
            var record = JObject.Parse(recordJson);

            Assert.AreEqual(id, record["id"]);
            Assert.AreEqual(type, record["type"]);
            Assert.AreEqual(value, record["value"]);
            Assert.IsTrue(JValue.DeepEquals(JObject.Parse(tags), JObject.Parse(record["tags"].ToString())));
        }

        [TestMethod]
        public async Task TestGetRecordWorksForNotFoundRecord()
        {
            var ex = await Assert.ThrowsExceptionAsync<WalletItemNotFoundException>(() =>
               NonSecrets.GetRecordAsync(wallet, type, id, optionsEmpty)
           );
        }
    }
}
